flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/plan/rules/physical/batch/DynamicPartitionPruningRule.java

```diff
@@ -138,30 +137,26 @@ private static List<Integer> getAcceptedFieldIndices(
                             .map(i -> factScan.getRowType().getFieldNames().get(i))
                             .collect(Collectors.toList());
         }
+        List<String> acceptedFilterFieldsInSource =
+                ((SupportsDynamicFiltering) tableSource).listAcceptedFilterFields();
 
-        List<String> acceptedFields =
-                ((SupportsDynamicFiltering) tableSource).applyDynamicFiltering(candidateFields);
-
-        if (acceptedFields == null) {
-            return new ArrayList<>();
-        }
-
-        for (String field : acceptedFields) {
-            if (!candidateFields.contains(field)) {
-                throw new TableException(
-                        String.format(
-                                "Field: %s does not exist in the given fields: %s, "
-                                        + "please verify the applyDynamicFiltering method in connector: %s",
-                                field, candidateFields, tableSource.asSummaryString()));
+        List<String> acceptedFilterFields = new ArrayList<>();
+        for (String candidateField : candidateFields) {
+            if (acceptedFilterFieldsInSource.contains(candidateField)) {
+                acceptedFilterFields.add(candidateField);
             }
         }
```

Comment:
extract a common method which can be reused by DynamicPartitionPruningRule

Reply:
> extract a common method which can be reused by DynamicPartitionPruningRule

Done!